docs/governance/rfc/005-python-implementation.md

```diff
@@ -0,0 +1,146 @@
+Adding an official implementation will help the Connect community focus its efforts.
+
+## Anticipated Complications
+
+
+The Python ecosystem has a big split between asynchronous and synchronous I/O, which will complicate the implementation story.
+Supporting both is important; Python applications tend to commit to one of the two sides of the split and have difficulty using the other.
+
+Further complicating matters is the Python community is divided regarding the right asynchronous implementation.
+[asyncio](https://docs.python.org/3/library/asyncio.html) is the standard library's answer, but you can find adherents of [trio](https://trio.readthedocs.io/en/stable/) and other asynchronous event loops.
+
+Python's immature support for http/2 is also a major challenge, particularly for servers.
+
+## Non-goals
+
+In order to make incremental progress, we explicitly cross off certain features in the initial pass at a Python implementation:
+
+1. *No cross-protocol support*: this will only be a client and server of the Connect protocol, not gRPC or grpc-web.
+2. *No full-duplex streams*: http/2 transport is hard to support on both client and server; half-duplex is fine though
+
+### Justification for not supporting gRPC on the server
+
+gRPC is a poor match to Python's most commonly used asynchronous server framework, ASGI.
+A correct gRPC implementation requires control over very low-level details of the connection state and http/2 transport.
+These details are not exposed by ASGI, so supporting them would prevent this library's servers from being ASGI applications.
+
+That's an unacceptable trade-off to make because ASGI interoperability is important for integrating connect servers into larger applications, like those running on Django, Starlette or FastAPI.
+
+### Justification for not supporting gRPC on the client
+
+Implementing a correct gRPC client again requires low-level control over the transport stack.
+The connect client would need to accept an extremely low-level and generic transport, even when contacting connect servers, just to support gRPC.
+
+This is more acceptable than the server case, but still unpleasant.
+Users will expect that they can pass an httpx.Client or similar to configure instrumentation, authentication, and so on.
+If we use a low-level library like hyper-h2 to implement the client, we'd likely need to build a bespoke abstraction supporting features like authentication which would be quite a bit more scope to cover.
+
+### Justification for not supporting grpc-web
+
+The grpc-web project lacks a standalone specification.
+It is defined [only in relation to gRPC](https://github.com/grpc/grpc/blob/6c7e2a94f99747950397258a8de005a3d90210a1/doc/PROTOCOL-WEB.md).
+
+As a result, most Connect implementations of grpc-web do so by treating it as a special case of gRPC.
+But without a gRPC implementation to base off of, a Python Connect library would need to implement grpc-web "from scratch" which is much more complex.
```

Comment:
gRPC-Web actually looks a **lot** like the Connect stream protocol. If the implementation supports streaming, this doesn't feel like the right justification to exclude gRPC-Web.